src/Codeception/Coverage/Subscriber/LocalServer.php

```diff
@@ -148,18 +149,52 @@ function ($h) {
 
     protected function startCoverageCollection($testName)
     {
-        $cookie = [
+        $value = [
             'CodeCoverage'        => $testName,
             'CodeCoverage_Suite'  => $this->suiteName,
             'CodeCoverage_Config' => $this->settings['remote_config']
         ];
+        $value = json_encode($value);
+
         $this->module->amOnPage('/');
-        $this->module->setCookie(self::COVERAGE_COOKIE, json_encode($cookie));
+        $this->module->setCookie(self::COVERAGE_COOKIE, $value);
+
+        // putting in configuration ensures the cookie is used for all sessions of a MultiSession test
+
+        $cookies = $this->module->_getConfig('cookies');
+        if (!$cookies|| !is_array($cookies)) {
+            $cookies = [];
+        }
+
+        $cookie = null;
+        foreach ($cookies as &$cookie) {
```

Comment:
It results in $cookie being set to a reference to the cookie we're interested in changing. You're probably right and I should rewrite it to something simpler.